### PR TITLE
forbid range scan of bool in lookup

### PR DESCRIPTION
**Pull request by critical27**
Some prepare work to modify null encode in index.
1. forbid range scan of bool in lookup, because `IndexColumnHint` can not make range [start, end) work.
2. add more test case for index scan rule test

### CODE REVIEW

File: src/optimizer/OptimizerUtils.cpp [hunk 1]
```diff
@@ -338,7 +334,7 @@ Value OptimizerUtils::boundValueWithMax(const meta::cpp2::ColumnDef& col, const
             return Value(std::numeric_limits<double>::max());
         }
         case Value::Type::BOOL: {
-            return v;
+            return Value(true);
```
Comment (bright-starry-sky) [inline src/optimizer/OptimizerUtils.cpp:337]: I think , better move the BOOL to below of switch block. let's unify that with boundValueWithLT and boundValueWithGT . WDYT?

**Comment by bright-starry-sky** [comment]
Good job. Thanks take this.

File: src/optimizer/OptimizerUtils.cpp (hunk 1 above)
Comment (critical27) [reply src/optimizer/OptimizerUtils.cpp:337]: ok, make sense

**Review by bright-starry-sky — APPROVED** [review]
Well done.

**Review by dutor — APPROVED** [review]
Great

**Pull request merged**